### ISSUE TITLE
Is creating client lws instance( maybe by lws_client_connect_via_info() )   dynamically  supported ?

### ISSUE THREAD

**Issue by cppbitman** [warmcat/libwebsockets#528]
Can I do lws_client_connect_via_info() to create lws instance 
from different thread than the one lws_service() loop runs ?

I do this because in my application I need to connect to lws server 
after lws_service() loop of client has started up already.
I've been looking for some api that can satisfy my demand. 

Can anybody help me?
Thanks a lot.

**Comment by lws-team** [comment]
Short answer, no.

Longer answer, lws fits into a style of programming that is event driven in a single thread.  That means it can dispense with various global overheads to do with blocking / threads / locking.

Instead of having your 'thing' that makes you want to connect in another thread, you would put it inside your event loop, from a timer, socket, file or other event.  Lws can work with external ('foreign') event loops with, eg, libuv.  Lws didn't invent this style, there are many event loop libraries.

So to get a simple ride, you should think about if the other things your app is doing can fit in the single event loop along with lws, everything serviced sequentially in one thread context.

**Comment by cppbitman** [comment]
@lws-team 
Thanks for your answer.

During the last two days, I have built libwebsockets with libuv supported, and the test-server-libuv example runs well.Then I make some modification on the test-server-libuv example.

 I created a lws_context instance with option LWS_SERVER_OPTION_EXPLICIT_VHOSTS | LWS_SERVER_OPTION_LIBUV, and called lws_libuv_run()  on that context instance after registering an uv_async_t callback with uv_async_init() in a thread.

Then I triggered the uv_async_t callback, in which I create a vhost listening on port 7681 in a different thread. Everything goes well so far. So I opened my browser and typed in "127.0.0.1:7681", but the browser didn't display the demo web with dumb-increment.

So I draw a conclusion that all vhosts on server side and all connections on client side must be created before lws_service() or lws_libuv_run() runs. Dynamically creating any of them is not supported.

Am I right? Thanks.
